components/devtools/actors/watcher.rs

```diff
@@ -429,4 +431,31 @@ impl WatcherActor {
             },
         }
     }
+
+    pub fn emit_will_navigate(
+        &self,
+        browsing_context_id: BrowsingContextId,
+        url: ServoUrl,
+        connections: &mut Vec<TcpStream>,
+    ) {
+        let msg = serde_json::json!({
```

Comment:
Can we create a new struct for this message and it's fields instead of using the json macro?